### PR TITLE
Issue #15015: Update CatchParameterName format to allow unnamed catch…

### LINKED ISSUE #15015: Add Check Support for Java 21 Unnamed Variables & Patterns Syntax: CatchParameterName
child of https://github.com/checkstyle/checkstyle/issues/14942
I have read check documentation: https://checkstyle.org/checks/naming/catchparametername.html#CatchParameterName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java                                                
public class Test {
    void test() {
        try {
            int x = 1/0;
        } catch (Exception _) {

        }
    }
}
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
         <module name="CatchParameterName"/>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:5:28: Name '_' must match pattern '^(e|t|ex|[a-z][a-z][a-zA-Z]+)$'. [CatchParameterName]
Audit done.
Checkstyle ends with 1 errors.
PS D:\CS\test> 
```

Describe what you expect in detail.

`CatchParameterName` should allow unnamed pattern variables by default

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #15015: 

Diff Regression config: https://gist.githubusercontent.com/mahfouz72/afd57aa6e51ca161b7927acd959e5211/raw/696b257d7628c4a99328382b0878d6a07a46ff1a/check.xml

Diff Regression projects: https://gist.githubusercontent.com/mahfouz72/a3d0af030c8f5efd0d8a39f2c14750bc/raw/8ca3493f88bc1003db86b40015a4c2101ab7bc94/projects-to-test-on.properties

### CODE REVIEW

**Comment by mahfouz72** [comment]
Github, generate site

**Comment by mahfouz72** [comment]
Github, generate report

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Please update documentation (add a bullet point) to mention that we allow underscores for unnamed variables, and update/remove the bullet point that mentions that we do not allow symbols.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java [hunk 1]
```diff
@@ -35,16 +35,16 @@
  * <li>allows {@code e} abbreviation (suitable for exceptions end errors)</li>
  * <li>allows {@code ex} abbreviation (suitable for exceptions)</li>
  * <li>allows {@code t} abbreviation (suitable for throwables)</li>
+ * <li>allows {@code _} for unnamed catch parameter</li>
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java:38]:
```suggestion
 * <li>allows {@code _} for unnamed catch parameters</li>
```
Other items are plural, let's be consistent

**Review by nrmancuso — CHANGES_REQUESTED** [review]
minor item, last from me:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java (hunk 1 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java]: done

**Review by nrmancuso — APPROVED** [review]
Good stuff.

**Comment by mahfouz72** [comment]
Github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java [hunk 2]
```diff
@@ -35,16 +35,16 @@
  * <li>allows {@code e} abbreviation (suitable for exceptions end errors)</li>
  * <li>allows {@code ex} abbreviation (suitable for exceptions)</li>
  * <li>allows {@code t} abbreviation (suitable for throwables)</li>
+ * <li>allows {@code _} for unnamed catch parameters</li>
  * <li>prohibits numbered abbreviations like {@code e1} or {@code t2}</li>
  * <li>prohibits one letter prefixes like {@code pException}</li>
  * <li>prohibits two letter abbreviations like {@code ie} or {@code ee}</li>
- * <li>prohibits any other characters than letters</li>
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java:41]: If we are going to remove this then there needs to be something equivalent that mentions the rest of the prohibitions not mentioned above. This statement covered those before.

For example, we don't say `$` is prohibited, nor underscores (excluding `_` alone) nor numbers. We say numbered abbreviations above, but its not just abbreviations, it is all numbers.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java (hunk 2 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java:41]: prohibits any other characters than letters and underscore?

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java:41]: > prohibits any other characters than letters and underscore?

The concern with just saying "underscore" is you have to make sure they understand underscore with anything else is disallowed. It is just a single `_` which is allowed.

Truthfully, I think original wording is fine. We say first we allow `_`. We can say we don't allow anything other than letters. It should be understood it is top to bottom order. Allows overrides the prohibits.

Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/CatchParameterNameCheck.java]: > Truthfully, I think original wording is fine. We say first we allow `_`. We can say we don't allow anything other than letters. It should be understood it is top to bottom order. Allows overrides the prohibits.

make sense. Original wording is restored

**Review by rnveach — APPROVED** [review]

**Pull request merged**